### PR TITLE
Control Flow improvements:

### PR DESCRIPTION
**Pull request by arturobernalg**
* Collapse  if conditions.
* Simplify if conditions.
* Remove duplicate code.

### CODE REVIEW

File: core/helper-tools/src/main/java/esa/mo/helpertools/helpers/HelperMisc.java [hunk 1]
```diff
@@ -338,11 +338,8 @@ public static boolean isStringAttribute(Attribute obj) throws IllegalArgumentExc
     if (shortFormPart == 15) { // String
       return true;
     }
-    if (shortFormPart == 18) { // URI
-      return true;
-    }
-
-    return false;
+      // URI
+      return shortFormPart == 18;
```
[the remark this answers is not in the document]
Comment (TanguySoto) [reply core/helper-tools/src/main/java/esa/mo/helpertools/helpers/HelperMisc.java:342]: @arturobernalg can you please fix and I'll merge

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java [hunk 2]
```diff
@@ -3312,23 +3312,6 @@ public CommandResult runGenericMethodForCommand(CommandDescriptor c)
                 result.append(calcNMEAChecksum(result.toString()));
               }
             }
-          } else if (inputSentence.endsWith("GPGRS")) {
```
Comment (TanguySoto) [inline mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java:3315]: @arturobernalg could you please explain to me why this disappears completely?

**Review by TanguySoto — CHANGES_REQUESTED** [review]

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java (hunk 2 above)
Comment (arturobernalg) [reply mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java:3315]: HI @TanguySoto 
easy, it's duplicated. check 
https://github.com/esa/nanosat-mo-framework/blob/338f182e13749efb553dcea00a1011754dba96c4/mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/threading/SimulatorNode.java#L3298

File: core/helper-tools/src/main/java/esa/mo/helpertools/helpers/HelperMisc.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (arturobernalg) [reply core/helper-tools/src/main/java/esa/mo/helpertools/helpers/HelperMisc.java]: damn. god spot. Sorry
Fixed,
TY

[the remark this answers is not in the document]
Comment (TanguySoto) [reply core/helper-tools/src/main/java/esa/mo/helpertools/helpers/HelperMisc.java]: Really sorry but the comment is still weirdly indented haha

[the remark this answers is not in the document]
Comment (TanguySoto) [reply core/helper-tools/src/main/java/esa/mo/helpertools/helpers/HelperMisc.java]: Hi @arturobernalg,
I'm not sure why you mean if you look at the line 341 on the file itself I still see a weird indentation https://github.com/esa/nanosat-mo-framework/blob/4d1323efbd4c46d41054842f2c3f685df54d4434/core/helper-tools/src/main/java/esa/mo/helpertools/helpers/HelperMisc.java#L341

**Pull request closed without merging**